edge-modules/TwinTester/TwinState.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TwinTester
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class TwinState
+    {
+        public int ReportedPropertyUpdateCounter;
+        public int DesiredPropertyUpdateCounter;
+        public string TwinETag;
+        public DateTime LastTimeOffline;
+
+        public TwinState(int reportedPropertyUpdateCounter, int desiredPropertyUpdateCounter, string twinETag, DateTime lastTimeOffline)
+        {
+            this.ReportedPropertyUpdateCounter = reportedPropertyUpdateCounter;
+            this.DesiredPropertyUpdateCounter = desiredPropertyUpdateCounter;
+            this.TwinETag = twinETag;
+            this.LastTimeOffline = lastTimeOffline;
```

Comment:
Where/how is the LastTimeOffline being used? Where/how is it set?

Reply:
LastTimeOffline is used [here](https://github.com/and-rewsmith/iotedge/blob/andsmi/longhaul-twin-dm/edge-modules/TwinTester/TwinOperator.cs#L102) to delay the validation of desired properties which have not reached the device due to being offline. It is [set](https://github.com/and-rewsmith/iotedge/blob/andsmi/longhaul-twin-dm/edge-modules/TwinTester/TwinOperator.cs#L206) in the reported properties validation loop if the device cannot getTwin() from the cloud.